spec/models/order_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe Order, type: :model do
+  describe "validates" do
+    it do
+      is_expected.to validate_presence_of(:phone)
+      is_expected.to validate_presence_of(:address)
+
+      is_expected.to define_enum_for(:status).with_values(%i(wait_confirm delivering completed canceled))
+    end
+  end
+
+  describe "associations" do
+    it do
+      is_expected.to have_many(:order_items)
+      is_expected.to have_many(:products).through(:order_items)
+      is_expected.to belong_to(:user)
+    end
+  end
+
+  describe "callback" do
+    let!(:product) { FactoryBot.create(:product, price: 20) }
+    
+    let(:order) do
+      order = FactoryBot.build(:order)
+      order.order_items.build(product: product, quantity: 2)
+      order.save
+      order
+    end
+
+    describe "#calculate_total_price" do
+      it { expect(order.total_price).to eq(40.0) }
+    end
+
+    describe "#update_sold_product" do
+      context "when update order's status is completed" do
+        it { expect { order.completed! }.to change { order.products.first.sold }.by(2) }
+      end
+
+      context "when update order's status is't completed" do
+        it { expect { order.delivering! }.not_to change { order.products.first.sold } }
+      end
+    end
+  end
+
+  describe "scope" do
+    before do
+      @order_previous_month = FactoryBot.create :order, created_at: Time.zone.now - 1.month
+      @order_this_month = FactoryBot.create :order, created_at: Time.zone.now
+    end
+
+    it { expect(described_class.all).to eq(described_class.all.order(id: :desc)) }
+
+    it do
+      records = described_class.this_month
+      
+      expect(records).to include(@order_this_month)
+      expect(records).not_to include(@order_previous_month)
+    end
+  end
```

Comment:
Đọc thêm về let và let! anh Tâm hỏi á
```suggestion
  describe "scopes" do
    describe "::this_month" do
      subject { described_class.this_month }
      
      let(:order) { create :order, created_at: created_at }
      context "when in this month" do ... end
      context "when not in this month" do ... end
    end
  end
```

Reply:
@tamtt-1031 :)) phần này bửa mạnh vào nha a